src/devices/DCMotor/DCMotor.cs

```diff
@@ -41,58 +41,130 @@ public virtual void Dispose()
+
+            return new DCMotor2PinNoEnable(speedControlChannel, -1, null);
+        }
+
+        /// <summary>
+        /// Creates DCMotor instance which allows to control speed in one direction.
+        /// </summary>
+        /// <param name="speedControlPin">Pin used to control the speed of the motor with software PWM</param>
+        /// <param name="controller">GPIO controller related to the pin</param>
+        /// <returns>DCMotor instance</returns>
+        public static DCMotor Create(int speedControlPin, GpioController controller = null)
+        {
+            if (speedControlPin == -1)
+                throw new ArgumentOutOfRangeException(nameof(speedControlPin));
+
+            controller = controller ?? new GpioController();
+            return new DCMotor2PinNoEnable(
+                new SoftwarePwmChannel(speedControlPin, 50, 0.0, controller: controller),
+                -1,
+                controller);
+        }
+
+        /// <summary>
+        /// Creates DCMotor instance which allows to control speed in both directions.
+        /// </summary>
+        /// <param name="speedControlChannel">PWM channel used to control the speed of the motor</param>
+        /// <param name="directionPin">Pin used to control the direction of the motor</param>
+        /// <param name="controller">GPIO controller related to the pin</param>
+        /// <returns>DCMotor instance</returns>
+        public static DCMotor Create(PwmChannel speedControlChannel, int directionPin, GpioController controller = null)
+        {
+            if (speedControlChannel == null)
+                throw new ArgumentNullException(nameof(speedControlChannel));
+
+            if (directionPin == -1)
+                throw new ArgumentOutOfRangeException(nameof(directionPin));
+
+            return new DCMotor2PinNoEnable(speedControlChannel, directionPin, controller);
+        }
+        
+        /// <summary>
+        /// Creates DCMotor instance which allows to control speed in both directions.
+        /// </summary>
+        /// <param name="speedControlPin">Pin used to control the speed of the motor with software PWM</param>
+        /// <param name="directionPin">Pin used to control the direction of the motor</param>
+        /// <param name="controller">GPIO controller related to the pins</param>
+        /// <returns>DCMotor instance</returns>
+        public static DCMotor Create(int speedControlPin, int directionPin, GpioController controller = null)
+        {
+            if (speedControlPin == -1)
+                throw new ArgumentOutOfRangeException(nameof(speedControlPin));
+
+            if (directionPin == -1)
+                throw new ArgumentOutOfRangeException(nameof(directionPin));
+
+            controller = controller ?? new GpioController();
+            return new DCMotor2PinNoEnable(
+                new SoftwarePwmChannel(speedControlPin, 50, 0.0, controller: controller),
```

Comment:
Same question about 50 hertz here. Should that be configurable?

Reply:
I can add a constant but it is configurable if you pass your own SoftwarePwmChannel instance